### ISSUE TITLE
Crossplatform: macOS

### ISSUE THREAD

**Issue by Co2p** [hainproject/hain#201]
Just tried compiling on a mac and the basic functionality seems to work.. (apart from that it's looking for the wrong filetypes it the wrong folders)

![screen shot 2016-11-03 at 15 02 28](https://cloud.githubusercontent.com/assets/6750851/19969093/89fe7212-a1d7-11e6-8211-fe85a5e245f9.png)

**Comment by staff0rd** [comment]
The dev branch includes extension specific filtering thanks to #192, however you are correct in that both these extensions and paths default to windows values, as [seen here](https://github.com/appetizermonster/hain/blob/develop/app/main/plugins/hain-plugin-filesearch/preferences.json).

If you can't make a PR to add this functionality, please suggest which paths and extensions would be considered default for mac os, perhaps myself or someone else can make a PR.

**Comment by Co2p** [comment]
I did try modifying the code, it still couldn't find anything however (undoubtedly I missed something somewhere).
binary paths for macs are:
- /Applications
- ~Applications (Users/ *user_name* /Applications)

Applications have the \*.app extension **however** they are not strictly speaking files but folders that launch the binary located inside, so usually the binary located here is launched:

`MyApp.app/Contents/MacOS/MyApp
`

Basically depends on how electron launches applications.. will probably need some work.
I don't really have time to dig deep at the moment, but would be really nice to have a cross platform plugin-powered application launcher 😄

**Comment by cpriest** [comment]
On the Mac platform, does it have PATH and PATHEXT environment variables?

**Comment by appetizermonster** [comment]
<img width="842" alt="hain-macos" src="https://cloud.githubusercontent.com/assets/6244462/22626914/d805d276-ebfa-11e6-9a51-b9669728b520.png">

Finally, it's almost finished.

**Comment by appetizermonster** [comment]
this issue has resolved with the version `0.6.2`
